src/QualificationRunner.Core/Services/QualificationRunner.cs

```diff
@@ -401,8 +401,30 @@ private SimulationParameterSwap mapSimulationParameter(SimulationParameterRef si
       private IReadOnlyList<SimulationPlot> retrieveProjectPlots(dynamic reportConfiguration) =>
          GetListFrom<SimulationPlot>(reportConfiguration.Plots?.AllPlots);
 
-      private IReadOnlyList<Input> retrieveInputs(dynamic reportConfiguration) =>
-         GetListFrom<Input>(reportConfiguration.Inputs);
+      private IReadOnlyList<Input> retrieveInputs(dynamic qualificationPlan)
+      {
+         var sections = retrieveSections(qualificationPlan);
+         var inputs = GetListFrom<Input>(qualificationPlan.Inputs);
+         foreach (var input in inputs)
+         {
+            input.SectionLevel = getSectionLevel(sections, input.SectionId);
+         }
+
+         return inputs;
+      }
+
+      private int? getSectionLevel(IReadOnlyList<dynamic> sections, int sectionId, int currentLevel = 1)

```

Comment:
This is not the most optimal way of retrieve the level. But the number of input is fairly limited and we will cost this function only a few times. Good enough for now